ui/library/src/components/thumbnails/Thumbnail.tsx

```diff
@@ -0,0 +1,63 @@
+import classnames from 'classnames';
+import * as React from 'react';
+
+import { PageRenderContext } from '../../context/PageRenderContext';
+import { ScrollContext } from '../../context/ScrollContext';
+import { getMaxVisibleElement } from '../../utils/MaxVisibleElement';
+import { Nullable } from '../types/utils';
+
+type Props = {
+  pageNumber: number;
+};
+
+export const Thumbnail: React.FunctionComponent<Props> = ({ pageNumber }: Props) => {
+  const { pageRenderStates, buildObjectURLForPage, getObjectURLForPage } =
+    React.useContext(PageRenderContext);
+  const { isPageVisible, scrollToPage, visiblePageRatios } = React.useContext(ScrollContext);
+
+  const [maxVisiblePageNumber, setMaxVisiblePageNumber] = React.useState<Nullable<string>>(null);
+
+  const objectURL = getObjectURLForPage({ pageNumber });
+
+  React.useEffect(() => {
+    if (visiblePageRatios.size !== 0) {
+      const max = getMaxVisibleElement(visiblePageRatios);
+      if (max) {
+        setMaxVisiblePageNumber(max.toString());
+      }
+    }
+  }, [visiblePageRatios]);
+
+  const hasPageVisible =
+    maxVisiblePageNumber &&
+    parseInt(maxVisiblePageNumber) === pageNumber &&
+    isPageVisible({ pageNumber });
+
+  React.useEffect(() => {
+    buildObjectURLForPage({ pageNumber });
+  }, [pageNumber, pageRenderStates]);
+
+  const onClick = React.useCallback(
+    event => {
+      event.preventDefault();
+      scrollToPage({ pageNumber });
+    },
+    [pageNumber, scrollToPage]
+  );
+
+  return (
+    <a
+      aria-label={`scroll to page ${pageNumber}`}
+      onClick={onClick}
+      className={classnames(
+        'reader__thumbnail',
+        { 'reader__thumbnail--has-page-image': objectURL },
+        { 'reader__thumbnail--no-page-image': !objectURL },
+        { 'reader__thumbnail--has-page-visible': hasPageVisible },
+        { 'reader__thumbnail--no-page-visible': !hasPageVisible }
+      )}
+      data-page-number={pageNumber}>
+      <img className="reader__thumbnail__image" src={objectURL || undefined} />
```

Comment:
`objectURL` is nullable. Maybe you can write like below
```suggestion
      {!!objectURL && <img className="reader__thumbnail-image" src={objectURL} />}
```

Reply:
@yensung if we change to this it will cause this lovely thing
![image](https://user-images.githubusercontent.com/84343285/191854385-e3ff2bb6-d150-4c4b-bb9c-cb5d4f5ae1f4.png)